hibernate-core/src/main/java/org/hibernate/Interceptor.java

```diff
@@ -52,7 +53,7 @@
  *
  * @author Gavin King
  */
-public interface Interceptor {
+public interface Interceptor extends StatementInspector {
```

Comment:
Related to above comment, does `Interceptor` itself need to extend `StatementInspector`?  I cannot remember the details, but I would have expected a delegation here (`EmptyStatementInspector#inspect` to `Interceptor#onPrepareStatement` on the Session's interceptor) if there was no explicit `StatementInspector`